bbid.py

```diff
@@ -1,34 +1,40 @@
 #!/usr/bin/env python3
-import os, sys, urllib.request, re, threading, posixpath, urllib.parse, argparse, random, socket, time, hashlib, pickle, signal, imghdr
+import os, sys, urllib.request, urllib.error, re, threading, posixpath, urllib.parse, argparse, random, socket, time, hashlib, pickle, signal, imghdr
 
 #config
 output_dir = './bing' #default output dir
 adult_filter = True #Do not disable adult filter by default
 pool_sema = threading.BoundedSemaphore(value = 20) #max number of download threads
 bingcount = 35 #default bing paging
+limit = 10 #default nb results limit
 socket.setdefaulttimeout(2)
 
-in_progress = tried_urls = []
+in_progress = {}
+tried_urls = []
 image_md5s = {}
 urlopenheader={ 'User-Agent' : 'Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0'}
 def download(url,output_dir):
-	if url in tried_urls:
+	search_id = output_dir.split('/')[-1]
+	if search_id not in in_progress:
+		in_progress[search_id] = []
+	if url in tried_urls or len(os.listdir(output_dir)) + len(in_progress[search_id]) > limit:
```

Comment:
I am not asking you to fix it, because I don't see a good way how, but I need to warn that this is not completely reliable: If there are 11 images available, limit is 10, 9 images were downloaded and 1 image is in progress, this will skip download of 11th image. However, 10th image might not be saved, because it doesn't pass validation constraints (e.g. it might not be valid image), resulting in only 9 images saved.